### DISCUSSION TITLE
front-end(UX, UI)와 back-end의 독립적인 설계

### DISCUSSION THREAD

**Discussion by baesy0** [digital-idea/OpenPipelineIO#1756] [General]
### 배경
이번 이슈에 대한 코드리뷰(#1755)를 할 때, Jason이 UX,UI라는 측면과 javascript의 측면을 고려해서 코드를 짰다고 한다.
이 설명을 들었을 때, 그렇다면 이 코드에 들어간 설계는 전체 js코드에 적용되는 것이 좋은 것 아닌가하는 생각이 들었다.
더불어, 지금까지 js부분을 어떻게 짜는게 좋을 것인가에 대한 고민이나 토론이 거의 없었다는 사실도 상기되었다.

### 우려되는 부분
단발성으로 이슈를 처리하면서 그 때 그 때 생각나는 아이디어로 코드를 짜면
- 레거시 코드가 생기기 쉽고,
- 설계의 일관성을 잃으면서 가독성이 떨어지고, 이해하기 힘든 코드가 되기 쉬울 것 같다.

### 제안
한 번쯤 전체적인 설계를 고민해보고 진행을하면
- 나중에 설계가 바뀌더라도 기존 코드에 일관성이 있기 때문에 설계를 바꾸기 쉬울 것 같다.
- 프론트앤드와 백앤드가 독립적으로 코드를 짤 수 있는 환경을 만들면, 조직이 확대되거나 분업이 일어날 때 편리할 것 같다.
- 프론트앤드 안에서의 분업도 더 편리할 것 같다.

**Comment by khw7096** [comment]
- 설계를 고민하기 위해서 실제로 한번은 돌아가는 레거시 코드가 필요하다.
- 그 코드를 바탕으로 좋은 구조를 고민해볼 수 있다.
- 백엔드는 RestAPI 만 작성하고 RestAPI를 통해 가지고 온 자료구조에 UX에 필요한 object 값을 추가해서 브라우저에 렌더링하는 방식은 어떠한 라이브러리를 사용하더라도 구조 이해에 좋겠다고 생각한다.
- 추후 UX 개발자가 투입된다면 도입 및 고려해볼 수 있는 방식을 생각날 때 마다 나열해보기
    - 자체 UX 라이브러리 구축: 개인적으로 선호. 추후 특정 이론이나 영향을 받을 때 특이점을 만들 수 있기 때문이다.
    - React
    - Vue
